pageserver/src/tenant_mgr.rs

```diff
@@ -104,6 +105,64 @@ struct Tenant {
     local_timelines: HashMap<ZTimelineId, Arc<DatadirTimelineImpl>>,
 }
 
+impl Tenant {
+    fn get_local_timeline_with_load(
+        &mut self,
+        timeline_id: ZTimelineId,
+    ) -> anyhow::Result<Arc<DatadirTimelineImpl>> {
+        if let Some(page_tline) = self.local_timelines.get(&timeline_id) {
+            Ok(Arc::clone(page_tline))
+        } else {
+            let page_tline = self
+                .load_local_timeline(timeline_id)
+                .with_context(|| format!("Failed to load local timeline {timeline_id}"))?;
+            self.local_timelines
+                .insert(timeline_id, Arc::clone(&page_tline));
+            Ok(page_tline)
+        }
+    }
+
+    fn load_local_timeline(
+        &mut self,
+        timeline_id: ZTimelineId,
+    ) -> anyhow::Result<Arc<DatadirTimeline<LayeredRepository>>> {
+        let inmem_timeline = self.repo.get_timeline_load(timeline_id).with_context(|| {
+            format!("Inmem timeline {timeline_id} not found in tenant's repository")
+        })?;
+        let repartition_distance = self.repo.get_checkpoint_distance() / 10;
+        let mut ancestor_logical_size: Option<usize> = None;
+        if let Some(ancestor_id) = inmem_timeline.get_ancestor_timeline_id() {
+            if let Ok(ancestor_pgdir) = self.get_local_timeline_with_load(ancestor_id) {
```

Comment:
nit: same with `pgdir`, maybe this should be changed to `ancestor_tline`

Reply:
Once again - there are two different things, ancestor timeline 'LayeredTime' and ancestor `DatadirTimelineImpl`. Just what to somehow distinguish them by variable name. May be bad idea, especially in case of using IDe which shows variable types.